constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,133 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		want    string
+	}{
+		{
+			name:    "no rewrite for ref is sub of externs",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			want:    "data.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.lib.util",
+			want:    "data.data.lib.lib.util",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, []string{"data.ext", "data.inventory"})
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			ref, err := ast.ParseRef(tt.imports)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			wantRef, err := ast.ParseRef(tt.want)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			if got := r.rewriteDataRef(ref); !got.Equal(wantRef) {
+				t.Errorf("rewriteDataRef() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAddEntryPointParseErrors(t *testing.T) {
+	tcs := []struct {
+		name      string
+		path      string
+		src       string
+		wantError error
+	}{
+		{
+			name: "add lib.rego path",
+			path: "data.stuff.foolib",
+			src: RegoSrc("lib.rego", Body(`test_ok {
```

Comment:
Use the raw rego rather than helper methods for this (src should just be assigned a string rather than the result of a method) - as-is the test case hides what the literal rego source is.

Reply:
Updated.